File: company-implementation/src/main/java/telran/employees/WageEmployee.java
package telran.employees;

import org.json.JSONObject;

public class WageEmployee extends Employee{
    private int wage;
    private int hours;
    public WageEmployee(){}
    public WageEmployee(long id, int basicSalary, String department, int wage, int hours) {
        super(id, basicSalary, department);
        this.wage = wage;
        this.hours = hours;
    }
    @Override
    public int computeSalary() {
        return super.computeSalary() + wage * hours;
    }
    @Override
     protected void fillJSON(JSONObject jsonObj) {
      super.fillJSON(jsonObj);
      jsonObj.put("wage", wage);
      jsonObj.put("hours", hours);
     }
     @Override
     protected void setObject(JSONObject jsonObj) {
         super.setObject(jsonObj);
         wage = jsonObj.getInt("wage");
         hours = jsonObj.getInt("hours");
      }
      public int getHours() {
          return hours;
      }
      public int getWage() {
          return wage;
      }
}

File: Node-Project/src/WageEmployee.mjs
import Employee from "./Employee.mjs";
export default class WageEmployee extends Employee{
    #wage;
    #hours
    constructor(id, department, basicSalary,  wage = 0, hours=0) {
        super(id, department, basicSalary);
        this.#wage = wage;
        this.#hours = hours;
    }
    getWage() {
        return this.#wage;
    }
    getHours() {
        return this.#hours
    }
    computeSalary () {
        return super.computeSalary() + this.#hours * this.#wage
    }
}